bento_drop_box_service/backends/local.py

```diff
@@ -42,23 +42,32 @@ async def _get_directory_tree(
                 entry_path_stat = entry_path.stat()
 
                 rel_path = (f"/{sub_path_str}" if sub_path_str else "") + f"/{entry}"
+                is_directory = await aiofiles.ospath.isdir(entry_path)
 
-                if not ((await aiofiles.ospath.isdir(entry_path)) or self.is_passing_filter(entry, include, ignore)):
+                if not (is_directory or self.is_passing_filter(entry, include, ignore)):
                     # If neither of these conditions is true, we should skip this entry in the tree
                     continue
 
+                # recurse on directories
+                if is_directory:
+                    contents = await self._get_directory_tree(
+                        root_path, (*sub_path, entry), level=level + 1, ignore=ignore, include=include
+                    )
+
+                    # if using ignore or include, skip empty directories
+                    if not contents and bool(ignore or include):
+                        continue
+
                 entries.append(
                     {
                         "name": entry,
                         "filePath": str(entry_path),  # Actual path on file system
                         "relativePath": rel_path,  # Path relative to root of drop box (/)
                         **(
                             {
-                                "contents": await self._get_directory_tree(
-                                    root_path, (*sub_path, entry), level=level + 1, ignore=ignore, include=include
-                                ),
+                                "contents": contents,
```

Comment:
I find conditionally-declared variables to trip me up when I read code. maybe contents could start as None or something? although that might trip up some type-checker/linter since entries is not supposed to have `"contents": None`

Reply:
either that, or create an `entry` variable which gets populated with a `content` field if `is_directory`, otherwise gets populated with the file entries, and then appended at the end.